### ISSUE TITLE
double free or corruption (out) with nightly build 485 recompiled on rpi3 armhf

### ISSUE THREAD

**Issue by Fantu** [FreeRDP/FreeRDP#4429]
On raspbian based on debian stretch I rebuilt nightly build 485 trying to check if solves freerdp crash on audio mode 0 with some usb headset.
Build fails, solved disabling address sanitizer also this time (I suppose also armhf need to be excluded by this option in debian/rules).
After built successfull I'm unable to use it, with any options gave me:
*** Error in `/opt/freerdp-nightly/bin/xfreerdp': double free or corruption (out): 0x7217e8d0 ***

I also installed debug symbols and run it with gdb:
```
/opt/freerdp-nightly/bin/xfreerdp /u:u1 /v:w10-3 /bpp:16
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/arm-linux-gnueabihf/libthread_db.so.1".
[New Thread 0x72a211a0 (LWP 13727)]
[16:45:33:128] [13724:13727] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
Password: 
[16:45:38:419] [13724:13727] [INFO][com.freerdp.gdi] - Local framebuffer format  PIXEL_FORMAT_BGRX32
[16:45:38:419] [13724:13727] [INFO][com.freerdp.gdi] - Remote framebuffer format PIXEL_FORMAT_RGB16
[New Thread 0x71d9e1a0 (LWP 13728)]
[New Thread 0x7159e1a0 (LWP 13729)]
[New Thread 0x70d9e1a0 (LWP 13730)]
[New Thread 0x7059e1a0 (LWP 13731)]
[16:45:38:531] [13724:13727] [INFO][com.winpr.clipboard] - initialized POSIX local file subsystem
[New Thread 0x6efed1a0 (LWP 13732)]
*** Error in `/opt/freerdp-nightly/bin/xfreerdp': double free or corruption (out): 0x7217e8d0 ***

Thread 2 "xfreerdp" received signal SIGABRT, Aborted.
[Switching to Thread 0x72a211a0 (LWP 13727)]
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
51	../sysdeps/unix/sysv/linux/raise.c: File o directory non esistente.
(gdb) bt full
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
        set = {__val = {0, 0, 1913843200, 1913843200, 1913885896, 8121, 0, 0, 
            0, 0, 1913885640, 8145, 1961955464, 1923220664, 1913819736, 
            1923220677, 1923220412, 1, 1913818448, 0, 1913885488, 1962926840, 
            0, 0, 1923220677, 1996483816, 16, 1913819688, 0, 1923221080, 
            1962743800, 1913821280}}
        pid = <optimized out>
        tid = <optimized out>
#1  0x76961824 in __GI_abort () at abort.c:89
        save_stage = 2
        act = {__sigaction_handler = {sa_handler = 0x0, sa_sigaction = 0x0}, 
          sa_mask = {__val = {520291095, 209, 0, 0, 891312, 1995087712, 0, 
              1994387868, 1996483816, 0, 0, 1406996736, 0, 0, 864000, 868328, 
              1, 1923220764, 1923220760, 1989221416, 864000, 1989216184, 0, 0, 
              711, 0, 1, 1923220824, 1, 98, 7, 7}}, sa_flags = 25, 
          sa_restorer = 0x72a20590}
        sigs = {__val = {32, 0 <repeats 31 times>}}
#2  0x7699af78 in __libc_message (do_abort=do_abort@entry=2, 
    fmt=<optimized out>) at ../sysdeps/posix/libc_fatal.c:175
        ap = {__ap = 0x72a2068c}
        fd = 25
        on_2 = <optimized out>
        list = <optimized out>
---Type <return> to continue, or q <return> to quit---
        nlist = <optimized out>
        cp = <optimized out>
        written = 120
#3  0x769a1ad4 in malloc_printerr (action=<optimized out>, 
    str=0x76a54068 "double free or corruption (out)", ptr=<optimized out>, 
    ar_ptr=<optimized out>) at malloc.c:5049
        buf = "7217e8d0"
        cp = <optimized out>
        ar_ptr = <optimized out>
        ptr = <optimized out>
        str = 0x76a54068 "double free or corruption (out)"
        action = <optimized out>
#4  0x769a2514 in _int_free (av=0x76a70794 <main_arena>, p=0x7217e8c8, 
    have_lock=<optimized out>) at malloc.c:3905
        size = <optimized out>
        fb = <optimized out>
        nextchunk = <optimized out>
        nextsize = <optimized out>
        nextinuse = <optimized out>
        prevsize = <optimized out>
        bck = <optimized out>
        fwd = <optimized out>
        errstr = <optimized out>
---Type <return> to continue, or q <return> to quit---
        locked = <optimized out>
        __func__ = "_int_free"
#5  0x76e9e5c8 in ?? () from /usr/lib/arm-linux-gnueabihf/libX11.so.6
No symbol table info available.
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
```
About (# 5) I already installed libx11-6-dbgsym but seems don't use it.

If you need more info/tests tell me and I'll do, thanks for any reply and sorry for my bad english.

**Comment by akallabeth** [comment]
Fixed such an issue today, already upstream

**Comment by bmiklautz** [comment]
@Fantu as @akallabeth said that should already be fixes on master with #4421. Either try to build master or wait  for this nights nightly.
